src/textual/widgets/_markdown.py

```diff
@@ -0,0 +1,790 @@
+    """A table header Markdown block."""
+
+    DEFAULT_CSS = """
+
+    """
+
+
+class MarkdownTD(MarkdownBlock):
+    """A table data Markdown block."""
+
+    DEFAULT_CSS = """
+
+    """
+
+
+class MarkdownBullet(Widget):
+    """A bullet widget."""
+
+    DEFAULT_CSS = """
+    MarkdownBullet {
+        width: auto;
+    }
+    """
+
+    symbol = reactive("●​ ")
+
+    def render(self) -> Text:
+        return Text(self.symbol)
+
+
+class MarkdownListItem(MarkdownBlock):
+    """A list item Markdown block."""
+
+    DEFAULT_CSS = """
+    MarkdownListItem {
+        layout: horizontal;
+        margin-right: 1;
+        height: auto;
+    }
+
+    MarkdownListItem > Vertical {
+        width: 1fr;
+        height: auto;
+    }
+    """
+
+    def __init__(self, bullet: str) -> None:
+        self.bullet = bullet
+        super().__init__()
+
+    def compose(self) -> ComposeResult:
+        bullet = MarkdownBullet()
+        bullet.symbol = self.bullet
+        yield bullet
+        yield Vertical(*self.blocks)
+
+        self.blocks.clear()
+
+
+class MarkdownFence(MarkdownBlock):
+    """A fence Markdown block."""
+
+    DEFAULT_CSS = """
+    MarkdownFence {
+        margin: 1 0;
+        overflow: auto;
+        width: 100%;
+        height: auto;
+        max-height: 20;
+    }
+
+    MarkdownFence > * {
+        width: auto;
+    }
+    """
+
+    def __init__(self, code: str, lexer: str) -> None:
+        self.code = code
+        self.lexer = lexer
+        super().__init__()
+
+    def compose(self) -> ComposeResult:
+        yield Static(
+            Syntax(
+                self.code,
+                lexer=self.lexer,
+                word_wrap=False,
+                indent_guides=True,
+                padding=(1, 2),
+                theme="material",
+            ),
+            expand=True,
+            shrink=False,
+        )
+
+
+HEADINGS = {
+    "h1": MarkdownH1,
+    "h2": MarkdownH2,
+    "h3": MarkdownH3,
+    "h4": MarkdownH4,
+    "h5": MarkdownH5,
+    "h6": MarkdownH6,
+}
+
+NUMERALS = " ⅠⅡⅢⅣⅤⅥ"
+
+
+class Markdown(Widget):
+    DEFAULT_CSS = """
+    Markdown {
+        height: auto;
+        margin: 0 4 1 4;
+        layout: vertical;
+    }
+    .em {
+        text-style: italic;
+    }
+    .strong {
+        text-style: bold;
+    }
+    .s {
+        text-style: strike;
+    }
+    .code_inline {
+        text-style: bold dim;
+    }
+    """
+    COMPONENT_CLASSES = {"em", "strong", "s", "code_inline"}
+
+    def __init__(
+        self,
+        markdown: str | None = None,
+        *,
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+    ):
+        """A Markdown widget.
+
+        Args:
+            markdown: String containing Markdown or None leave blank for now. Defaults to None.
+            name: The name of the button.
+            id: The ID of the button in the DOM.
+            classes: The CSS classes of the button.
```

Comment:
These 3 lines talk about `button` when this is the `Markdown` widget. Possible copy/paste-o?